### PR TITLE
Allow Lambda invocation context to be injected

### PR DESCRIPTION
**Pull request by i906**
With this PR, I can inject the Lambda invocation context into jobs.

```php
<?php

namespace App\Jobs;

use Bref\Context\Context;
use Illuminate\Bus\Queueable;
use Illuminate\Contracts\Queue\ShouldQueue;
use Illuminate\Foundation\Bus\Dispatchable;
use Illuminate\Queue\InteractsWithQueue;
use Illuminate\Queue\SerializesModels;

class LambdaTest implements ShouldQueue
{
    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;

    protected $work = 0;

    public function handle(Context $context)
    {
        $shouldPause = false;

        while (!$this->isWorkDone() && !$shouldPause) {
            $this->doWork();

            // check deadline
            $remaining = $context->getRemainingTimeInMillis();
            $shouldPause = $remaining < 1500;
        }

        if (!$this->isWorkDone()) {
            // pause and requeue
        } else {
            // cleanup
        }
    }

    protected function doWork()
    {
        $this->work++;
    }

    protected function isWorkDone()
    {
        return $this->work >= 100;
    }
}

```

### CODE REVIEW

File: src/BrefServiceProvider.php [hunk 1]
```diff
@@ -64,5 +67,26 @@ public function register(): void
         // to avoid errors. If you want to actively use the cache, it will be best to use
         // the dynamodb driver instead.
         Config::set('cache.stores.file.path', '/tmp/storage/framework/cache');
+
+        $this->registerInvocationContext();
+    }
+
+    protected function registerInvocationContext(): void
+    {
+        $this->app->singleton(Context::class, function () {
```
Comment (mnapoli) [inline src/BrefServiceProvider.php:76]: Thanks for the PR, I don't think it should be a singleton in the container:

- the context is not a service, like the request
- if Laravel is kept alive between invocations then the context will be wrong

Comment (i906) [reply src/BrefServiceProvider.php:76]: I see. Would it be better if there is another class to handle this? e.g. 

```php
public function handle(\Bref\Lambda $bref)
{
    /** @var Context $context */
    $context = $bref->getContext();
}

Comment (mnapoli) [reply src/BrefServiceProvider.php:76]: Not really (adding a new class would be overkill). Injecting the Context in `handle(Context $context)` makes sense, we just shouldn't register this as a service.

I'm not sure how it's doable however.

**Comment by mnapoli** [comment]
Closing as this PR is from 2020 and the Queues integration will radically change in https://github.com/brefphp/laravel-bridge/pull/94

**Pull request closed without merging**